lib/beaker/host/freebsd/pkg.rb

```diff
@@ -1,13 +1,28 @@
 module FreeBSD::Pkg
   include Beaker::CommandFactory
 
-  def install_package(name, cmdline_args = nil, opts = {})
-    cmdline_args ||= '-y'
-    execute("pkg install #{cmdline_args} #{name}", opts) { |result| result }
+  def shell_check_pkgng
+    # See manpage of pkg(8), the paragraph devoted to -N flag
+    # https://www.freebsd.org/cgi/man.cgi?query=pkg
+    'TMPDIR=/dev/null ASSUME_ALWAYS_YES=1 PACKAGESITE=file:///nonexist ' \
+    'pkg info -x "pkg(-devel)?\\$" > /dev/null 2>&1'
   end
 
-  def check_for_package(name, opts = {})
-    execute("pkg info #{name}", opts) { |result| result }
+  def shell_ifelse(cond, stmt_t, stmt_f)
+    "if #{cond}; then #{stmt_t}; else #{stmt_f}; fi"
   end
 
+  def install_package(package, cmdline_args = nil, opts = {})
+    cmd = shell_ifelse shell_check_pkgng,
+                       "pkg install #{cmdline_args || '-y'} #{package}",
+                       "pkg_add #{cmdline_args || '-r'} #{package}"
+    execute("/bin/sh -c '#{cmd}'", opts) { |result| result }
+  end
+
+  def check_for_package(package, opts = {})
+    cmd = shell_ifelse shell_check_pkgng,
+                       "pkg info #{package}",
+                       "pkg_info -Iqx #{package} 2> /dev/null || true"
```

Comment:
Oh my, I don't actually remember where it comes from, thought I've found it somewhere in beakers code. The problem is, that in case ``#{package}`` is not installed, ``pkg_info`` exits with exit code ``1`` and beaker exits with failure. Looks like I have to elaborate this a little bit.